core/src/test/java/org/apache/calcite/test/UdfTest.java

```diff
@@ -198,6 +235,19 @@
     assertThat(after, is(before + 4));
   }
 
+  @Test public void testUserDefinedFunctionWithArgumentAssignment() throws Exception {
+    final String sql = "select \"adhoc\".my_plus(x=>\"deptno\", y=>100) as p\n"
+        + "from \"adhoc\".EMPLOYEES";
+    final AtomicInteger c = Smalls.MyPlusFunction.INSTANCE_COUNT.get();
+    final int before = c.get();
+    withUdf().query(sql).returnsUnordered("P=110",
+        "P=120",
+        "P=110",
```

Comment:
Could you please add some negative cases?

Reply:
I have added more test cases for this PR.